discord/discordDeleteMessage.js

```diff
@@ -0,0 +1,96 @@
+module.exports = function (RED) {
+  var discordBotManager = require('./lib/discordBotManager.js');
+
+  function discordDeleteMessage(config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+    var configNode = RED.nodes.getNode(config.token);
+    discordBotManager.getBot(configNode).then(function (bot) {
+      node.on('input', function (msg) {
+        if (!msg.id || !msg.channel) {
+          node.error(`Either msg.id or msg.channel wasn't set`);
+          node.status({
+            fill: "red",
+            shape: "dot",
+            text: `msg.id and msg.channel are required`
+          })
+          return;
+        }
+
+        var msgId = msg.id;
+        if (msgId && typeof msgId !== 'string') {
+          if (msgId.hasOwnProperty('id')) {
+            msgId = msgID.id;
+          } else {
+            msgId = undefined;
+            node.error(`msg.Id needs to be either a string for the id or channel Object`);
+            node.status({
+              fill: "red",
+              shape: "dot",
+              text: `msg.id is not a string or Object`
+            })
+            return;
+          }
+        }
+
+        var channel = msg.channel;
+        if (channel && typeof channel !== 'string') {
+          if (channel.hasOwnProperty('id')) {
+            channel = channel.id;
+          } else {
+            channel = undefined;
```

Comment:
Since there's a return on line 48, clearing the `channel` local variable is superfluous